### DISCUSSION TITLE
当阅读模式并未符合你的期望时，如何进一步优化？

### DISCUSSION THREAD

**Discussion by Kenshin** [Kenshin/simpread#1554] [使用技巧]
> 建议使用方案二

# 方案一

只需要解决两个工具 [重新高亮](http://ksria.com/simpread/docs/#/重新高亮) 与 [隐藏任意元素](om/simpread/docs/#/隐藏任意元素) ，就可以实现这个诉求。

位置在：`进入阅读模式 → 右下角控制栏 → 动作`

![image](https://user-images.githubusercontent.com/81074/60070858-d47f5700-974a-11e9-9699-0cc9bbe226b1.png)

![image](https://user-images.githubusercontent.com/81074/132842175-449f5789-a5b5-4e15-ac46-6a759fc553da.png)

通过手动整理后的页面，方便导出 [各种生产力工具](http://ksria.com/simpread/docs/#/导出到生产力工具)

# 方案二（新版首选方案）

在原文选取任意内容，通过右键  → 将选中的内容生成阅读模式

<img width="584" alt="image" src="https://user-images.githubusercontent.com/81074/193535305-c8e1db68-2737-4c02-8772-7eb56bf06bce.png">

如果你的右键菜单没有这项的话，请在选项页 → 高级设定 → 右侧菜单开启

<img width="1282" alt="image" src="https://user-images.githubusercontent.com/81074/193535560-a22d0678-3e80-4212-9a97-e89a1209a06a.png">

**Comment by elnino2081** [comment]
<img width="196" alt="CleanShot 2022-02-28 at 14 12 58@2x" src="https://user-images.githubusercontent.com/19463793/155933517-27615845-d117-4b8e-ab10-adb245142aa8.png">

发现“重选高亮区域”无法结果的页面，根本不会出现这两个按钮

**Reply by Kenshin** [reply]
这是正常的，你想要什么效果？

**Reply by egon-g** [reply]
请问为什么经过我手动 高亮和隐藏元素的页面 ，当我保存到稍后读之后，又变成了没有经过调整的页面版式？

**Reply by Kenshin** [reply]
@e9on 

如果要实现你说的方式，需要配置知识库，这是 [教程](https://kb.simpread.pro/#/page/%E5%BB%BA%E7%AB%8B%E7%9F%A5%E8%AF%86%E5%BA%93)。

**Comment by filosfino** [comment]
@Kenshin Hello 我在 https://www.wikiwand.com/zh/%E6%B4%9E%E7%A9%B4%E5%A5%87%E6%A1%88 这个页面尝试移除 #section_navigation-section 这个元素，但在阅读模式下失去了 section 的 id attribute，导致这个规则无法生效，想了解下 id attribute 被移除是预期中的吗？
<img width="1126" alt="image" src="https://user-images.githubusercontent.com/22309511/177715386-3d9f0df2-0f34-40c1-8ffd-80a73ca14c50.png">
<img width="516" alt="image" src="https://user-images.githubusercontent.com/22309511/177715395-9aafb4bc-3269-41e3-a9f1-2e51ec1178ae.png">

**Reply by filosfino** [reply]
这是阅读模式下的 DOM 结构
<img width="536" alt="image" src="https://user-images.githubusercontent.com/22309511/177715578-d5af720d-7feb-402e-a713-017ea382f609.png">

**Reply by Kenshin** [reply]
> 导致这个规则无法生效，想了解下 id attribute 被移除是预期中的吗？

是的，进入阅读模式后，这些会被移除，这是个选项，位置在：选项页 → 高级设定 → 词法分析引擎

![image](https://user-images.githubusercontent.com/81074/177722598-df40895f-74c7-4c5b-89eb-a7374ff1f4fc.png)

进入阅读模式后需要使用隐藏元素来通过 xpath 删除。

![image](https://user-images.githubusercontent.com/81074/177722796-4f150c94-93af-460a-a82d-2cd281d7a112.png)

**Reply by filosfino** [reply]
成功了，谢谢哈 ：）

**Reply by Kenshin** [reply]
小事情，搞定就好。

**Comment by sysir** [comment]
虽然使用了各种方案，但是快捷键下一页还是会回到源网页 https://www.92dxs.cc/reader/78/78906/73836670.html

**Reply by Kenshin** [reply]
将 `92dxs.cc` 加入白名单，细节 https://github.com/Kenshin/simpread/discussions/2117

**Reply by dahuyouT** [reply]
方案二真香……

**Reply by Kenshin** [reply]
@Tdahuyou

哈哈，理论上，方案二可以解决一切问题。

😄

**Comment by hidoos** [comment]
太牛了，想给你打钱，请你喝咖啡。

**Reply by Kenshin** [reply]
哈，谢谢喜欢。

你可以 [**升级到高级账户**](https://simpread.pro/price) 或在 [**Chrome Webstore**](https://chrome.google.com/webstore/detail/simpread-reader-view/ijllcpnolfcooahcekpamkbidhejabll) 五星好评支持我~

💪  😀

**Comment by jacob2826** [comment]
通过方案一调整后的阅读模式，在使用导出到md时，看起来用的还是未调整的页面导出的。
同时，刷新页面并再次进入阅读模式，仍为未调整的页面。（这里我不确定插件是否会保存我的调整记录）

**Reply by Kenshin** [reply]
> 通过方案一调整后的阅读模式，在使用导出到md时，看起来用的还是未调整的页面导出的。

我没有遇到你的问题，**重新得到的正文再导出时，就是新的正文内容**，你可以再重新确认下。

> 刷新页面并再次进入阅读模式，仍为未调整的页面

这是预期行为，简悦有「记录功能」但并没有那么好用，所以暂时并没有记录上一次的框选规则。

**Comment by ONIONLYONE** [comment]
大佬，有没有办法实现批量选择元素进入阅读模式吗，比如粉笔题库网站我只要题目和解析，其他的元素都不要。我之前使用过一款爬虫插件web scraper有批量选择的功能，不知道简悦能不能实现，谢谢。

**Reply by Kenshin** [reply]
嗨，暂时不支持这种形式，不过想法不错，我会考虑的。

目前有个比较笨的办法，你可以使用 **方案二** 进入阅读模式，然后使用 [Live Editor](https://github.com/Kenshin/simpread/discussions/2852) 删除不需要的内容。

**Reply by ONIONLYONE** [reply]
> 嗨，暂时不支持这种形式，不过想法不错，我会考虑的。
> 
> 目前有个比较笨的办法，你可以使用 **方案二** 进入阅读模式，然后使用 [Live Editor](https://github.com/Kenshin/simpread/discussions/2852) 删除不需要的内容。

感谢大佬开发，已支持。我也有尝试方案一然后一个个隐藏元素，我从站点编辑器中发现这些元素都是相似的，因此觉得应该可以实现批量隐藏，如果能开发批量隐藏的插件，对我这种编程小白的帮助很大。
[[`/html[1]/div[1]/sr-read[1]/sr-rd-content[1]/div[1]/main[1]/section[1]/div[1]/app-fb-solution[1]/fb-ng-solution[1]/div[1]/article[1]/ul[1]/li[1]`]]
[[`/html[1]/div[1]/sr-read[1]/sr-rd-content[1]/div[1]/main[1]/section[1]/div[2]/app-fb-solution[1]/fb-ng-solution[1]/div[1]/article[1]/ul[1]/li[1]`]]
[[`/html[1]/div[1]/sr-read[1]/sr-rd-content[1]/div[1]/main[1]/section[1]/div[3]/app-fb-solution[1]/fb-ng-solution[1]/div[1]/article[1]/ul[1]/li[1]`]]
[[`/html[1]/div[1]/sr-read[1]/sr-rd-content[1]/div[1]/main[1]/section[1]/div[4]/app-fb-solution[1]/fb-ng-solution[1]/div[1]/article[1]/ul[1]/li[1]`]]
[[`/html[1]/div[1]/sr-read[1]/sr-rd-content[1]/div[1]/main[1]/section[1]/div[5]/app-fb-solution[1]/fb-ng-solution[1]/div[1]/article[1]/ul[1]/li[1]`]]

**Reply by Kenshin** [reply]
> 因此觉得应该可以实现批量隐藏，如果能开发批量隐藏的插件，对我这种编程小白的帮助很大。

emmm 可以考虑下。但批量隐藏使用的标识仅限某个具体的网页，比如你提到的需求。

但这个需求似乎比较小众，除了你提到的场景外，我没有想到其它可用场景。

从需求点触发，感觉不如：**批量选择元素进入阅读模式** 更具有使用场景。😂

PS：当然我完全可以考虑开发这个插件，只是希望可以找到更多使用场景。

**Reply by ONIONLYONE** [reply]
> > 因此觉得应该可以实现批量隐藏，如果能开发批量隐藏的插件，对我这种编程小白的帮助很大。
> 
> emmm 可以考虑下。但批量隐藏使用的标识仅限某个具体的网页，比如你提到的需求。
> 
> 但这个需求似乎比较小众，除了你提到的场景外，我没有想到其它可用场景。
> 
> 从需求点触发，感觉不如：**批量选择元素进入阅读模式** 更具有使用场景。😂
> 
> PS：当然我完全可以考虑开发这个插件，只是希望可以找到更多使用场景。

哈哈，我的需求其实就是批量选择相似元素进入阅读模式，比如题库里有90题，里面有题目，解析，来源，正确率等等。只选中这90题的“题目”和“解析”进入阅读模式。
因为目前的解决方案是隐藏，所以我给绕进去了，哈哈哈哈哈哈哈哈哈。

**Reply by Kenshin** [reply]
小事情，我研究下你的想法。😂 😎

**Comment by Lawrence-of-AnKing** [comment]
[中文马克思主义文库](https://www.marxists.org/chinese/maozedong/index.htm)这个网站不适配阅读模式，如：[论反对日本帝国主义的策略](https://www.marxists.org/chinese/maozedong/marxist.org-chinese-mao-19351227.htm)

即使使用“将选中的内容生成阅读模式”，也无法正常调用快捷键，因为“手动框选”没有除快捷键外的进入方法（可能有，但我没找到），这使得手册中的“手动框选”->“加入适配列表”的方式走不通，“选中生成”后也没看到“加入适配列表”的选项，也找不到怎么加入“站点集市”中的“待决站点”。

我希望能为这个网站增加基本的适配，并且能自动进入，该怎么做？

另，即使选中后生成阅读模式，也有很多换行问题，如下：

![image](https://github.com/user-attachments/assets/24489351-07ae-4569-b806-37ddf8a56d30)

最新Chrome + Win11

**Reply by Kenshin** [reply]
嗨，在这个页面的最上方就有解决方案

<img width="459" alt="image" src="https://github.com/user-attachments/assets/3613ad96-dac7-4a84-b61c-6970d8e9516e">

**Reply by Lawrence-of-AnKing** [reply]
我已经找到办法了，在`站点` - `重新选项高亮区域` - 选择想进入阅读模式的内容即可，不得不说这个描述很让人疑惑，明明是选择内容进入阅读模式，为什么是“重新选项高亮区域”？

不过那个错误换行问题还是没法解决，这种应该怎么办？

**Reply by Kenshin** [reply]
> 那个错误换行问题还是没法解决，这种应该怎么办？

只能单独适配这个页面。（我会考虑适配这个网站的~

**Comment by fucao** [comment]
两种办法都不行啊，进入阅读模式后，会有一个满屏的原网站的内容，向下滚动就是阅读模式的样子。
<img width="2852" height="1524" alt="image" src="https://github.com/user-attachments/assets/d3e92eca-38d3-4330-bac8-2ff8a89943b3" />

<img width="2869" height="1564" alt="image" src="https://github.com/user-attachments/assets/03057f3a-7ec7-48a3-b777-c46de1786b53" />

**Reply by fucao** [reply]
https://www.fangzhenxiu.com/post/2591689/  网址是这个

**Reply by Kenshin** [reply]
我这边没问题

<img width="1162" height="594" alt="image" src="https://github.com/user-attachments/assets/a86797f9-0b2f-49f9-afcb-23ed53305a1f" />


<img width="1374" height="685" alt="image" src="https://github.com/user-attachments/assets/04aa572f-f58d-49bb-a1d2-6800ae952eb2" />
